doc/upgrades/3.x.x to 4.x.x.md

```diff
@@ -0,0 +1,44 @@
+# Migration guide for versions 4.x.x -> 5.x.x
+
+The most important change this release includes is the replacement of the HTTP library from a native one to
+OkHttp which allows SDK to
+
+ - Support the HTTP2 version of the HTTP protocol.
+ - Support proxies that do not use only basic authentication method. For details on creating custom proxy authenticators and an example of
+NTLM proxy authentication, see [here](https://github.com/box/box-java-sdk/blob/kb/ok-http/doc/configuration.md#custom-proxy-authenticator).
+ - 
+# Configuration changes
+
+## MaxRequestAttempts
+The `MaxRequestAttempts` are removed from `com.box.sdk.BoxGlobalSettings` and replaced with `MaxRetryAttempts`. 
+However, if you have API connection stored with the deprecated `MaxRequestAttempts` value it will be restored into `MaxRetryAttempts`.
+
+# Removed deprecated methods and classes
+1. `com.box.sdk.BoxAPIRequest.BoxAPIRequest(java.net.URL, java.lang.String)`
+2. `com.box.sdk.BoxUser.moveFolderToUser` - this method is removed as this operation is not allowed by API. You can only transfer root folder to another user using `com.box.sdk.BoxUser.transferContent`
+
+# Replaced methods and classes
+1. `com.box.sdk.BoxDeveloperEditionAPIConnection` - removed all constructors that do not accept `com.box.sdk.IAccessTokenCache`. Use versions that accepts cache parameter
+2. `com.box.sdk.BoxDeveloperEditionAPIConnection#getAppUserConnection` - replaced with com.box.sdk.BoxDeveloperEditionAPIConnection#getUserConnection
+3. `com.box.sdk.BoxEvent.getType` - replaced with `com.box.sdk.BoxEvent.getEventType`
+4. `com.box.sdk.BoxEvent.Type` - replaced with `com.box.sdk.BoxEvent.EventType`
+5. `com.box.sdk.BoxItem.createSharedLink` - removed methods that do not use `com.box.sdk.sharedlink.BoxSharedLinkRequest`, please ones that use it
```

Comment:
`removed methods that do not use...please ones that use it`
I guess this sentence is missing some words. Please also check this in other places